internal/paascharm/go.tmpl

```diff
@@ -0,0 +1,127 @@
+// Code generated by paascharmgen; DO NOT EDIT.
+package {{ .PackageName }}
+
+type CharmConfig struct {
+     ConfigOptions ConfigOptions
+     ProxyConfig   ProxyConfig
+     {{- if .Integrations }}
+         Integrations Integrations
+     {{- end }}
+}
+
+type ConfigOptions struct {
+     BaseURL string `env:"APP_BASE_URL"`
+     Port int `env:"APP_PORT"`
+     MetricsPort *int `env:"APP_METRICS_PORT"`
+     MetricsPath *string `env:"APP_METRICS_PATH"`
+     SecretKey *string `env:"APP_SECRET_KEY"`
+     {{- if .Options }}
+         UserConfigOptions
```

Comment:
Shouldn't it be `UserConfigOptions UserConfigOptions` ?

Reply:
I have used an embedded struct on purpose, so all the config options are together in one struct (like in charmcraft.yaml) and also because the user does not have to type an extra `.UserConfigOptions`. WDYT?